DependencyInjection/Configuration.php

```diff
@@ -42,18 +42,37 @@ public function getConfigTreeBuilder()
             ->children()
                 ->scalarNode('disable_csrf_role')->defaultNull()->end()
                 ->arrayNode('access_denied_listener')
-                    ->useAttributeAsKey('name')
-                    ->prototype('boolean')->end()
+                    ->canBeEnabled()
+                    ->beforeNormalization()
+                        ->ifArray()->then(function($v) { if (!empty($v) && empty($v['formats'])) { unset($v['enabled']); $v = array('enabled' => true, 'formats' => $v); } return $v; })
+                    ->end()
+                    ->fixXmlConfig('format', 'formats')
+                    ->children()
+                        ->scalarNode('service')->defaultNull()->end()
+                        ->arrayNode('formats')
+                            ->useAttributeAsKey('name')
+                            ->prototype('boolean')->end()
+                        ->end()
+                    ->end()
                 ->end()
                 ->scalarNode('unauthorized_challenge')->defaultNull()->end()
-                ->scalarNode('param_fetcher_listener')->defaultFalse()
+                ->arrayNode('param_fetcher_listener')
                     ->validate()
                         ->ifNotInArray($this->forceOptionValues)
                         ->thenInvalid('The param_fetcher_listener option does not support %s. Please choose one of '.json_encode($this->forceOptionValues))
                     ->end()
+                    ->canBeEnabled()
+                    ->children()
+                        ->scalarNode('service')->defaultNull()->end()
+                    ->end()
                 ->end()
                 ->scalarNode('cache_dir')->cannotBeEmpty()->defaultValue('%kernel.cache_dir%/fos_rest')->end()
-                ->scalarNode('allowed_methods_listener')->defaultFalse()->end()
+                ->arrayNode('allowed_methods_listener')
+                    ->canBeEnabled()
+                    ->children()
+                        ->scalarNode('service')->defaultNull()->end()
```

Comment:
looks like you only care about it being empty or no. So a boolean ``disable_listener`` might be better as you don't care about the actual id.